tests/ci/integration/netty_tcnative_patch/netty-tcnative-parent-2.0.62.Final.patch

```diff
@@ -0,0 +1,1276 @@
+ // BoringSSL we can just ignore this call. In the future we should remove the method all together.
+-#ifndef OPENSSL_IS_BORINGSSL
++#if !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_IS_AWSLC)
+     char err[ERR_LEN];
+ 
+     if (tcn_SSL_use_certificate_chain_bio(ssl_, b, skipfirst) < 0)  {
+@@ -2075,7 +2075,7 @@ TCN_IMPLEMENT_CALL(void, SSL, setCertificateChainBio)(TCN_STDARGS, jlong ssl,
+         ERR_clear_error();
+         tcn_Throw(e, "Error setting certificate chain (%s)", err);
+     }
+-#endif // OPENSSL_IS_BORINGSSL
++#endif // defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+ }
+ 
+ TCN_IMPLEMENT_CALL(jlong, SSL, loadPrivateKeyFromEngine)(TCN_STDARGS, jstring keyId, jstring password)
+@@ -2139,7 +2139,7 @@ TCN_IMPLEMENT_CALL(jlong, SSL, parseX509Chain)(TCN_STDARGS, jlong x509ChainBio)
+ {
+     BIO *cert_bio = J2P(x509ChainBio, BIO *);
+ 
+-#ifdef OPENSSL_IS_BORINGSSL
++#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+     STACK_OF(CRYPTO_BUFFER) *chain = sk_CRYPTO_BUFFER_new_null();
+     CRYPTO_BUFFER *buffer = NULL;
+     char *name = NULL;
+@@ -2149,14 +2149,14 @@ TCN_IMPLEMENT_CALL(jlong, SSL, parseX509Chain)(TCN_STDARGS, jlong x509ChainBio)
+ #else
+     X509* cert = NULL;
+     STACK_OF(X509) *chain = NULL;
+-#endif // OPENSSL_IS_BORINGSSL
++#endif // defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+ 
+     char err[ERR_LEN];
+     unsigned long error;
+ 
+     TCN_CHECK_NULL(cert_bio, x509ChainBio, 0);
+ 
+-#ifdef OPENSSL_IS_BORINGSSL
++#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+     while (PEM_read_bio(cert_bio, &name, &header, &data, &data_len)) {
+ 
+         OPENSSL_free(name);
+@@ -2174,15 +2174,15 @@ TCN_IMPLEMENT_CALL(jlong, SSL, parseX509Chain)(TCN_STDARGS, jlong x509ChainBio)
+     chain = sk_X509_new_null();
+     while ((cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL)) != NULL) {
+         if (sk_X509_push(chain, cert) <= 0) {
+-#endif // OPENSSL_IS_BORINGSSL
++#endif // defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+ 
+             tcn_Throw(e, "No Certificate specified or invalid format");
+             goto cleanup;
+         }
+ 
+-#ifndef OPENSSL_IS_BORINGSSL
++#if !defined(OPENSSL_IS_BORINGSSL) && !defined(OPENSSL_IS_AWSLC)
+         cert = NULL;
+-#endif // OPENSSL_IS_BORINGSSL
++#endif // defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+     }
+ 
+     // ensure that if we have an error its just for EOL.
+@@ -2202,25 +2202,25 @@ TCN_IMPLEMENT_CALL(jlong, SSL, parseX509Chain)(TCN_STDARGS, jlong x509ChainBio)
+ cleanup:
+     ERR_clear_error();
+ 
+-#ifdef OPENSSL_IS_BORINGSSL
++#if defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
+     sk_CRYPTO_BUFFER_pop_free(chain, CRYPTO_BUFFER_free);
+ #else
+     sk_X509_pop_free(chain, X509_free);
+     X509_free(cert);
+-#endif // OPENSSL_IS_BORINGSSL
++#endif // defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
```

Comment:
I don't get some of these if/def's we have `sk_X509_pop_free` and `X509_free`

Reply:
This is due to the code in netty-tcnative leveraging the `TLS_with_buffers_method` when constructing the `SSL_CTX` with `SSL_CTX_new`. This is a feature that BoringSSL added that Netty is taking advantage of to avoid usage of the more expensive X509 structure. This retains the certificate in a `CRYPTO_BUFFER`, thus alternative functions need to be used for the differing types. See our [porting guide](https://github.com/aws/aws-lc/blob/main/PORTING.md?plain=1#L287).